esb/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/BAMMediatorTransformer.java

```diff
@@ -22,68 +21,62 @@
 import org.apache.commons.lang.StringUtils;
 import org.apache.synapse.endpoints.Endpoint;
 import org.apache.synapse.mediators.base.SequenceMediator;
-import org.eclipse.core.runtime.Assert;
 import org.eclipse.emf.ecore.EObject;
+import org.wso2.carbon.mediator.bam.config.stream.StreamConfiguration;
 import org.wso2.developerstudio.eclipse.gmf.esb.BAMMediator;
 import org.wso2.developerstudio.eclipse.gmf.esb.EsbNode;
 import org.wso2.developerstudio.eclipse.gmf.esb.persistence.TransformationInfo;
 import org.wso2.developerstudio.eclipse.gmf.esb.persistence.TransformerException;
-import org.wso2.carbon.mediator.bam.config.stream.StreamConfiguration;
 
-public class BAMMediatorTransformer extends AbstractEsbNodeTransformer{
+public class BAMMediatorTransformer extends AbstractEsbNodeTransformer {
+
+    public void transform(TransformationInfo information, EsbNode subject) throws TransformerException {
+        information.getParentSequence().addChild(createBAMMediator(subject, information));
+
+        doTransform(information, ((BAMMediator) subject).getOutputConnector());
+    }
 
-	public void transform(TransformationInfo information, EsbNode subject) throws TransformerException {
-		information.getParentSequence().addChild(
-				createBAMMediator(subject, information));
+    public void createSynapseObject(TransformationInfo info, EObject subject, List<Endpoint> endPoints) {
```

Comment:
It should be implemented since BAMMediatorTransformer extends from AbstractEndpointTransformer. I will put @Override annotation to that method.
Thanks